Sources/SwifterSwift/SwiftStdlib/KeyedDecodingContainer.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  KeyedDecodingContainer.swift
+//  SwifterSwift
+//
+//  Created by Francesco Deliro on 23/10/2019.
+//  Copyright © 2019 SwifterSwift
+//
+
+#if canImport(Foundation)
+import Foundation
+#endif
+
+public extension KeyedDecodingContainer {
+
+    #if canImport(Foundation)
+    /// SwifterSwift: Try to decode a Bool as Int then String before decoding as Bool.
+    ///
+    /// - Parameter key: Key.
+    /// - Returns: Decoded Bool value.
+    /// - Throws: Decoding error.
+    func decodeBoolAsIntOrString(forKey key: Key) throws -> Bool {
+        if let intValue = try? decode(Int.self, forKey: key) {
+            return (intValue as NSNumber).boolValue
+        } else if let stringValue = try? decode(String.self, forKey: key) {
+            return (stringValue as NSString).boolValue
+        } else {
+            return try decode(Bool.self, forKey: key)
+        }
+    }
+    #endif
+
+    #if canImport(Foundation)
+    /// SwifterSwift: Try to decode a Bool as Int then String before decoding as Bool if present.
+    ///
+    /// - Parameter key: Key.
+    /// - Returns: Decoded Bool value.
+    /// - Throws: Decoding error.
+    func decodeBoolAsIntOrStringIfPresent(forKey key: Key) throws -> Bool? {
+        if let intValue = try? decodeIfPresent(Int.self, forKey: key) {
+            return (intValue as NSNumber).boolValue
```

Comment:
Please replace this with return `return intValue == 1`

Reply:
As commented by @guykogus I preferred to use NSString and NSInteger's .boolValue properties, which look for one of several predefined values meaning true.

In this case quoting Apple Doc we have :

**NSNumber boolValue**
> A 0 value always means false, and any nonzero value is interpreted as true.

Here we can easily move to a swifty solution, but for the string I think it would be preferable to use boolValue, what do you think about? :)